src/Datadog.Trace.OpenTracing/HttpHeadersCodec.cs

```diff
@@ -31,39 +29,21 @@ public OpenTracingSpanContext Extract(object carrier)
                 {
                     traceIdHeader = keyVal.Value;
                 }
-            }
 
-            if (parentIdHeader == null)
-            {
-                throw new ArgumentException($"{HttpHeaderNames.ParentId} should be set.");
-            }
-
-            if (traceIdHeader == null)
-            {
-                throw new ArgumentException($"{HttpHeaderNames.TraceId} should be set.");
+                if (keyVal.Key.Equals(HttpHeaderNames.SamplingPriority, StringComparison.OrdinalIgnoreCase))
+                {
+                    samplingPriorityHeader = keyVal.Value;
+                }
             }
 
-            ulong parentId;
-            try
-            {
-                parentId = Convert.ToUInt64(parentIdHeader);
-            }
-            catch (FormatException)
-            {
-                throw new FormatException($"{HttpHeaderNames.ParentId} should contain an unsigned integer value");
-            }
+            ulong.TryParse(parentIdHeader, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parentId);
+            ulong.TryParse(traceIdHeader, NumberStyles.Integer, CultureInfo.InvariantCulture, out var traceId);
 
-            ulong traceId;
-            try
-            {
-                traceId = Convert.ToUInt64(traceIdHeader);
-            }
-            catch (FormatException)
-            {
-                throw new FormatException($"{HttpHeaderNames.TraceId} should contain an unsigned integer value");
-            }
+            var samplingPriority = int.TryParse(samplingPriorityHeader, out int samplingPriorityValue)
+                                                     ? (SamplingPriority?)samplingPriorityValue
+                                                     : null;
 
-            SpanContext ddSpanContext = new SpanContext(traceId, parentId);
+            SpanContext ddSpanContext = new SpanContext(traceId, parentId, samplingPriority);
```

Comment:
Same here, we probably want to avoid using this context if `traceId == 0`

Reply:
Added a check in e367b4e584aabd032eef50d0b3a911e4a1939699. The C# OpenTracing docs are not clear on whether we can return null here or not, but is seems that the Java implementation expects null, so I'm going with that.

Reply:
nice!